assert/colorize.go

```diff
@@ -0,0 +1,72 @@
+package assert
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"sync"
+)
+
+const (
+	expected_colon = "expected:"
+	actual_colon   = "actual  :"
+)
+
+var (
+	shouldColorize    bool
+	setShouldColorize sync.Once
+)
+
+func figureOutShouldColorize() bool {
+	// FIXME: The auto response should be true on terminals and false on
+	// not-terminals, but my attempts at checking os.Stdout() terminalness have
+	// all failed. Patches welcome!
+	autoResponse := false
+
+	hint := os.Getenv("TESTIFY_COLOR")
```

Comment:
Instead if defining a new env variably could we define this behavior based on the current environment?
Maybe with something like this: https://wiki.archlinux.org/title/Color_output_in_console#Enumerate_supported_colors

This is how people is doing it in js:
https://github.com/chalk/supports-color/blob/ee88ebcfca649233bb35c9b0db226059883a77b8/index.js#L46

Reply:
That JS function takes a `streamIsTTY` parameter:

```
function _supportsColor(haveStream, {streamIsTTY, sniffFlags = true} = {}) {
```

And I don't know how to figure that out :(

https://github.com/stretchr/testify/blob/c602661a25d7372ccd53be44c8dd6445068eb336/assert/colorize.go#L21-L25

... so I can't really do it like that unfortunately.

Reply:
And in the long run, somebody will want to override the default behavior, and then the environment variable is likely to be needed anyway.

Reply:
@GerardRodes is this OK or do you have some suggestion for how to improve it?